multinode-demo/start_nodes.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Format for the nodes file
+# node1 public-ip|node1 privte-ip
+# node2 public-ip|node2 privte-ip
+# ....
+# e.g.
+# 35.197.92.75|10.138.0.6
+# 35.185.232.222|10.138.0.3
+# 35.233.137.53|10.138.0.5
+
+nodes_file=$1
+remote_user=$2
+ssh_keys=$3
+
+declare -A ip_addr_array
```

Comment:
does bash 3.2 have associative arrays?

Reply:
Python does :-P

Reply:
3.2 doesn't seem to have associative arrays. I was using this script from my ubuntu VM. Let me find a better solution.

Reply:
Given there's no associative arrays, any suggestions on the data structure to use? I can use a simple array with alternative entries being public and private IP. I don't like this approach for it's ugliness and requirements on the user to know this.

Reply:
It won't need private IP once we change the full-node code to use public IP. We can wait for that change..

Reply:
without associative arrays, you could just to tuples in a flat array:
```
ip_addrs = ( 
    35.197.92.75 10.138.0.6
    35.185.232.222 10.138.0.3
                    ....
            )
```
then you'd read them by pairs:

``
for ((i=0;i<${#ip_addrs[*]};i+=2)); do
  pub=${ip_addrs[i]}
  priv=${ip_addrs[i+1]}
...
```